### PR TITLE
handle swagger ui if CLI flag is set

### PR DESCRIPTION
**Pull request by bogdan-rosianu**
- add `--start-swagger-ui` CLI flag for serving the Swagger UI as well
- if the flag is set, static serve the Swagger UI at root ( `http(s)://proxy-url` )
- add `openapi.json`

future steps:
- use libs to automatically generate the swagger files based on the endpoints' annotations

Testing procedures:
1. start the proxy without the `--start-swagger-ui` flag. it should work just as before
2. start the proxy with the `--start-swagger-ui` flag. 
2.1. check the rest of the endpoints, should work like before
2.2. check the root of the proxy and validate the Swagger documentation as much as possible

### CODE REVIEW

**Review by miiu96 — DISMISSED** [review]

**Review by gabi-vuls — DISMISSED** [review]
System test passed

File: cmd/proxy/config/swagger/openapi.json [hunk 1]
```diff
@@ -0,0 +1,3016 @@
...
+        }
+      }
+    },
+    "/block/{shard}/by-nonce/{nonce}?withTxs=true": {
+      "get": {
+        "tags": [
+          "block"
+        ],
+        "summary": "returns a block by nonce, with transactions included.",
+        "parameters": [
```
Comment (ssd04) [inline cmd/proxy/config/swagger/openapi.json:739]: here we need also a parameter for `shard`

File: cmd/proxy/config/swagger/openapi.json [hunk 2]
```diff
@@ -0,0 +1,3016 @@
...
+        }
+      }
+    },
+    "/block/{shard}/by-hash/{hash}": {
+      "get": {
+        "tags": [
+          "block"
+        ],
+        "summary": "returns a block by hash.",
+        "parameters": [
```
Comment (ssd04) [inline cmd/proxy/config/swagger/openapi.json:770]: here also

File: cmd/proxy/config/swagger/openapi.json [hunk 3]
```diff
@@ -0,0 +1,3016 @@
...
+                }
+              }
+            }
+          }
+        }
+      }
+    },
+    "/block/{shard}/by-hash/{hash}?withTxs=true": {
+      "get": {
+        "tags": [
```
Comment (ssd04) [inline cmd/proxy/config/swagger/openapi.json:797]: here also

File: cmd/proxy/config/swagger/openapi.json [hunk 4]
```diff
@@ -0,0 +1,3016 @@
...
+            }
+          },
+          "400": {
+            "description": "validation error"
+          }
+        }
+      }
+    },
+    "/proof/root-hash/:roothash/address/:address": {
+      "get": {
```
Comment (ssd04) [inline cmd/proxy/config/swagger/openapi.json:1356]: input param for address here, maybe update also description field

File: cmd/proxy/config/swagger/openapi.json [hunk 5]
```diff
@@ -0,0 +1,3016 @@
...
+        ],
+        "summary": "returns a block by hash.",
+        "parameters": [
+          {
+            "name": "hash",
+            "in": "path",
+            "description": "the hash to look after",
+            "required": true,
+            "schema": {
+              "type": "integer"
```
Comment (ssd04) [inline cmd/proxy/config/swagger/openapi.json:777]: type `string` ?

File: cmd/proxy/config/swagger/openapi.json [hunk 6]
```diff
@@ -0,0 +1,3016 @@
...
+        ],
+        "summary": "returns a block by hash, with transactions included.",
+        "parameters": [
+          {
+            "name": "hash",
+            "in": "path",
+            "description": "the hash to look after",
+            "required": true,
+            "schema": {
+              "type": "integer"
```
Comment (ssd04) [inline cmd/proxy/config/swagger/openapi.json:808]: type `string`?

File: cmd/proxy/config/swagger/openapi.json [hunk 7]
```diff
@@ -0,0 +1,3016 @@
...
+            }
+          },
+          {
+            "name": "nonce",
+            "in": "path",
+            "required": true,
+            "description": "the nonce to look after",
+            "schema": {
+              "type": "string"
```
Comment (ssd04) [inline cmd/proxy/config/swagger/openapi.json:1334]: not sure about types here, interger for nonce and string for token?

File: cmd/proxy/config/swagger/openapi.json [hunk 8]
```diff
@@ -0,0 +1,3016 @@
...
+        ],
+        "summary": "generates the Merkle proof for the given address at a specific root hash",
+        "parameters": [
+          {
+            "name": "roothash",
+            "in": "path",
+            "description": "",
+            "required": true,
+            "schema": {
+              "type": "integer"
```
Comment (ssd04) [inline cmd/proxy/config/swagger/openapi.json:1368]: type `string`?

File: cmd/proxy/config/swagger/openapi.json (hunk 1 above)
Comment (bogdan-rosianu) [reply cmd/proxy/config/swagger/openapi.json:739]: added

File: cmd/proxy/config/swagger/openapi.json (hunk 3 above)
Comment (bogdan-rosianu) [reply cmd/proxy/config/swagger/openapi.json:797]: added

File: cmd/proxy/config/swagger/openapi.json (hunk 5 above)
Comment (bogdan-rosianu) [reply cmd/proxy/config/swagger/openapi.json:777]: fixed

File: cmd/proxy/config/swagger/openapi.json (hunk 6 above)
Comment (bogdan-rosianu) [reply cmd/proxy/config/swagger/openapi.json:808]: done

File: cmd/proxy/config/swagger/openapi.json (hunk 7 above)
Comment (bogdan-rosianu) [reply cmd/proxy/config/swagger/openapi.json:1334]: yup, changed

File: cmd/proxy/config/swagger/openapi.json (hunk 8 above)
Comment (bogdan-rosianu) [reply cmd/proxy/config/swagger/openapi.json:1368]: fixed

File: cmd/proxy/config/swagger/openapi.json (hunk 4 above)
Comment (bogdan-rosianu) [reply cmd/proxy/config/swagger/openapi.json:1374]: done

**Review by ssd04 — DISMISSED** [review]

**Review by gabi-vuls — DISMISSED** [review]

**Review by gabi-vuls — APPROVED** [review]

**Review by miiu96 — APPROVED** [review]

**Review by valentin-lup — APPROVED** [review]
Tested on MultiversX Public TestNet Proxy

**Pull request merged**